sqlstats/sqlstats_test.go

```diff
@@ -119,3 +121,62 @@ func TestCollect(t *testing.T) {
 		}
 	}
 }
+
+func TestTracerResetRace(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	tracer := &Tracer{}
+	db := sql.OpenDB(sqlite.Connector("file:"+t.TempDir()+"/test.db", nil, tracer))
+	defer db.Close()
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tx.Rollback()
+	if _, err := tx.ExecContext(ctx, "CREATE TABLE t (c);"); err != nil {
+		t.Fatal(err)
+	}
+	if err := tx.Commit(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Continually reset the tracer.
+	go func() {
+		for ctx.Err() == nil {
+			tracer.Reset()
+		}
+	}()
+
+	// Continually grab transactions and insert into the database.
+	errc := make(chan error, 1)
```

Comment:
We shouldn't need a buffer here, does the test not work without it? (If so I'd like to understand why, since AFAIK it should be fine)

Reply:
Buffer not required - I've removed it.